website/pages/api-docs/secret/key-management/index.mdx

```diff
@@ -0,0 +1,540 @@
+```
+
+## Rotate Key
+
+This endpoint rotates the version of a named key.
+
+| Method | Path                        |
+| :----- | :-------------------------- |
+| `PUT`  | `/keymgmt/key/:name/rotate` |
+
+### Parameters
+
+- `name` `(string: <required>)` – Specifies the name of the key to rotate.
+  This is provided as part of the request URL.
+
+### Sample Request
+
+```shell-session
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request PUT \
+    http://127.0.0.1:8200/v1/keymgmt/key/example-key/rotate
+```
+
+## Delete Key
+
+This endpoint deletes a named key. The key must be removed from all KMS providers that it's
+been distributed to and have `deletion_allowed` set to `true` in order to be deleted.
+
+| Method   | Path                  |
+| :------- | :-------------------- |
+| `DELETE` | `/keymgmt/key/:name`  |
+
+### Parameters
+
+- `name` `(string: <required>)` – Specifies the name of the key to delete.
+  This is provided as part of the request URL.
+
+### Sample Request
+
+```shell-session
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request DELETE \
+    http://127.0.0.1:8200/v1/keymgmt/key/example-key
+```
+
+## List KMS Providers of Key
+
+This endpoint returns a list of all KMS providers that the named key has been distributed to.
+Currently, a key can only be distributed to a single KMS provider.
+
+| Method | Path                     |
+| :----- | :----------------------- |
+| `LIST` | `/keymgmt/key/:name/kms` |
+
+### Parameters
+
+- `name` `(string: <required>)` – Specifies the name of the key.
+  This is provided as part of the request URL.
+
+### Sample Request
+
+```shell-session
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request LIST \
+    http://127.0.0.1:8200/v1/keymgmt/key/example-key/kms
+```
+
+### Sample Response
+
+```json
+{
+  "data": {
+    "keys": ["example-kms"]
+  }
+}
+```
+
+## Create/Update KMS Provider
+
+This endpoint creates or updates a KMS provider. If a KMS provider with the given `name`
+does not exist, it will be created. If the KMS provider exists, it will be updated with
+the given parameter values.
+
+| Method | Path                  |
+| :----- | :-------------------- |
+| `PUT`  | `/keymgmt/kms/:name`  |
+
+### Parameters
+
+- `name` `(string: <required>)` – Specifies the name of the KMS provider to create or update.
+  This is provided as part of the request URL.
+
+- `provider` `(string: <required>)` – Specifies the name of a KMS provider that's external to Vault.
+  Cannot be changed after creation. The following values are supported:
+  - `azurekeyvault` - KMS provider represents a named
+  [Azure Key Vault](https://docs.microsoft.com/en-us/azure/key-vault/) instance.
+
+- `key_collection` `(string: <required>)` – Refers to a location to store keys in the specified
+  provider. Cannot be changed after creation. The expected value for this parameter will differ
+  depending on the specified provider. The following values are expected for each provider:
```

Comment:
Oh actually its not quite the same.

`- azurekeyvault - KMS provider represents a named [Azure Key Vault]`
vs 
`- azurekeyvault - The name of an existing Azure Key Vault instance.`

Having said this, I think it might be a bit confusing since the first case is implying that `azurekeyvault` is one of the valid values to be provided, but for the second case the value for `key_collection` is some the actual instance _within_ Azure Key Vault, and not the value `azurekeyvault` itself.

Reply:
Good catch! I've updated this in f4e38e4 and 595d7f6.